pkg/sidecar/sidecar_test.go

```diff
@@ -0,0 +1,72 @@
+package sidecar
+
+import (
+	"context"
+	"github.com/alicebob/miniredis/v2"
+	"github.com/testground/sdk-go/runtime"
+	"github.com/testground/sdk-go/sync"
+	"math/rand"
+	"testing"
+	"time"
+)
+
+var TST_RUNPARAMS = runtime.RunParams{
+	TestPlan:               "sidecarPlan",
+	TestCase:               "sidecarCase",
+	TestRun:                "123",
+	TestInstanceCount:      1,
+	TestInstanceRole:       "sidecarRole",
+	TestGroupID:            "sidecarGroup",
+	TestGroupInstanceCount: 1,
+}
+
+func TestHandler(t *testing.T) {
+	r, err := miniredis.Run()
+	if err != nil {
+		t.Error(err)
+	}
+	defer r.Close()
+
+	runenv := runtime.NewRunEnv(TST_RUNPARAMS)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	client, err := sync.NewBoundClient(ctx, runenv)
+	if err != nil {
+		t.Error(err)
+	}
+
+	nw := NewMockNetwork()
+
+	time.Sleep(time.Duration(rand.Int()%10) * time.Second)
```

Comment:
What I am trying to do is avoid having multiple sidecars access the same redis keys. Will fix this.